tests/unit/manage/test_views.py

```diff
@@ -379,8 +379,10 @@ def test_change_primary_email(self, monkeypatch, db_request):
         user = UserFactory()
         old_primary = EmailFactory(primary=True, user=user)
         new_primary = EmailFactory(primary=False, verified=True, user=user)
+        email = EmailFactory(user=user)
```

Comment:
@waseem18 The issue is that we're adding an extra email here. Since we've already created the `old_primary` email above, and since `EmailFactory` defaults to making the created email primary as well, we had created two primary emails for the user in this test.

The way `User.email` works is that it assumes there might be multiple primary emails, and returns the first primary email it finds for the user (not totally why this is the case, but I'm guessing backwards compatibility with legacy).

In this test, most of the time, `user.email` would refer to this `email`, but sometimes (not sure why) they might be created out of order, so `user.email` would be `old_primary` instead, causing the test to fail.

The fix is in #3178: just create one primary email for the user, and test that email is the one that gets the message.

Reply:
Thank you so much for the detailed response and the PR @di

Reply:
No worries! Sorry that I missed it in review and you spent so much time trying to debug it.